extensions/sockets/src/main/java/org/mule/module/socket/api/connection/tcp/TcpRequesterConnection.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.socket.api.connection.tcp;
+
+import static java.lang.String.format;
+import static org.mule.module.socket.internal.SocketUtils.configureConnection;
+import org.mule.module.socket.api.client.TcpClient;
+import org.mule.module.socket.api.connection.RequesterConnection;
+import org.mule.module.socket.api.exceptions.UnresolvableHostException;
+import org.mule.module.socket.api.protocol.TcpProtocol;
+import org.mule.module.socket.api.socket.tcp.TcpClientSocketProperties;
+import org.mule.runtime.api.connection.ConnectionException;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of {@link RequesterConnection} for establishing TCP connections.
+ *
+ * @since 4.0
+ */
+public class TcpRequesterConnection extends AbstractTcpConnection implements RequesterConnection
+{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TcpRequesterConnection.class);
+    private final Socket socket;
+    private final TcpClientSocketProperties socketProperties;
+    private final InetSocketAddress address;
+
+    public TcpRequesterConnection(TcpClientSocketProperties socketProperties, TcpProtocol protocol, String host, Integer port) throws ConnectionException
+    {
+        super(host, port, protocol);
+        socket = new Socket();
+        address = getSocketAddress(host, port, socketProperties.getFailOnUnresolvedHost());
+        this.socketProperties = socketProperties;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public TcpClient getClient()
+    {
+        return new TcpClient(socket, protocol);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void disconnect()
+    {
+        try
+        {
+            LOGGER.debug("Closing TCP socket");
```

Comment:
I completely agree with this but the thing here is that the different Java socket classes doesn't share a common Interface even though they do have several common behaviours like for example socket.close()